### ISSUE TITLE
load balancing length delimited (prefixed) TCP messages

### ISSUE THREAD

**Issue by rkbalgi** [envoyproxy/envoy#3650]
Hey Guys, I wanted to know if  there a way to load balance TCP messages which are length delimited/prefixed (a two byte/4 byte etc). These kind of messages are common in the payments industry which use formats like ISO8583.

Example - a "hello world" would be sent as 000b68656c6c6f20776f726c64 (where 000b is the length prefix and the rest is hello world in ASCII). I would like to break down the message and then send it upstream to a cluster with a load balancer policy.

If such a feature is not available, any guidance to adding such support would be very helpful!

**Comment by mattklein123** [comment]
You could do this via custom filter.

**Comment by rkbalgi** [comment]
Thanks Matt! Would this be a good starting point? - https://github.com/envoyproxy/envoy-filter-example

**Comment by mattklein123** [comment]
Yup or other filters in https://github.com/envoyproxy/envoy/tree/master/source/extensions

**Comment by GustavoAdrianGimenez** [comment]
Hi @rkbalgi, How are you? Did you find some way to balance ISO 8583 traffic with envoy? can you share some example? Thank you!

**Comment by rkbalgi** [comment]
Hi @GustavoAdrianGimenez We ultimately chose to write our own simplified version of a proxy.
